### PR TITLE
Upgrade to SpiderMonkey 67

### PR DESCRIPTION
**Pull request by jdm**
Depends on https://github.com/servo/mozjs/pull/186.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/465)

### CODE REVIEW

File: src/jsglue.cpp [hunk 1]
```diff
@@ -31,6 +31,50 @@ typedef size_t(*GetSize)(JSObject *obj);
...
+  {
+    return mTraps.enqueuePromiseJob(mQueue, cx, promise, job, allocationSite, incumbentGlobal);
+  }
+
+  virtual bool empty() const {
+    return mTraps.empty(mQueue);
+  }
+
+  virtual void runJobs(JSContext* cx) {
+    MOZ_ASSERT("runJobs should not be invoked");
```
Comment (asajeffrey) [inline src/jsglue.cpp]: Hmm, this is a bit annoying, I can imagine it being triggered in future versions of SM/Servo.

File: src/jsglue.cpp [hunk 2]
```diff
@@ -31,6 +31,50 @@ typedef size_t(*GetSize)(JSObject *obj);
...
+  }
+
+private:
+  virtual js::UniquePtr<SavedJobQueue> saveJobQueue(JSContext* cx) {
+    MOZ_ASSERT("saveJobQueue should not be invoked");
+    return nullptr;
```
Comment (asajeffrey) [inline src/jsglue.cpp]: Ditto.

**Comment by jdm** [comment]
Retrying the single failed job.

**Pull request merged**